src/v2/coins/abstractUtxoCoin.js

```diff
@@ -1206,14 +1235,17 @@ class AbstractUtxoCoin extends BaseCoin {
         transactionBuilder.addOutput(krsFeeAddress, krsFee);
       }
 
-      const signedTx = this.signRecoveryTransaction(transactionBuilder, unspents, addressesById, !isKrsRecovery);
-
-      txInfo.transactionHex = signedTx.build().toBuffer().toString('hex');
-
-      try {
-        txInfo.tx = yield this.verifyRecoveryTransaction(txInfo);
-      } catch (e) {
-        console.log('Could not verify recovery transaction', e.message);
+      if (isUnsignedSweep) {
+        const txHex = transactionBuilder.buildIncomplete().toBuffer().toString('hex');
+        return this.formatForOfflineVault(txInfo, txHex);
+      } else {
+        const signedTx = this.signRecoveryTransaction(transactionBuilder, unspents, addressesById, !isKrsRecovery);
+        txInfo.transactionHex = signedTx.build().toBuffer().toString('hex');
+        try {
+          txInfo.tx = yield this.verifyRecoveryTransaction(txInfo);
+        } catch (e) {
+          console.log('Could not verify recovery transaction', e.message);
```

Comment:
Do you really want to continue here? If so, please leave a comment as to why you are not rethrowing (rethrowing should be the default, with an explanation as to why the error is being swallowed)

Reply:
hmm yeah kevin k wrote that a while back. i'll change it to throw.

Reply:
changing this broke a lot of tests because it calls an external "decodetx" api that we now need to nock. i can fix them